curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.english.md

```diff
@@ -23,12 +23,14 @@ Then fix the code so that the regex that you have created is checked against <co
 
 ```yml
 tests:
-  - text: Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Hello"</code>
-    testString: assert(myRegex.test('Hello'), 'Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Hello"</code>');
-  - text: Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Halo"</code>
-    testString: assert(myRegex.test('Halo'), 'Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Halo"</code>');
-  - text: Your regex <code>myRegex</code> should return <code>false</code> for the string <code>"Hell"</code>
-    testString: assert(!myRegex.test('Hell'), 'Your regex <code>myRegex</code> should return <code>false</code> for the string <code>"Hell"</code>');
+  - text: Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Franklin Roosevelt"</code>
+    testString: assert(myRegex.test('Franklin Roosevelt'), 'Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Franklin Roosevelt"</code>');
+  - text: Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Franklin D. Roosevelt"</code>
+    testString: assert(myRegex.test('Franklin D. Roosevelt'), 'Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Franklin D. Roosevelt"</code>');
+  - text: Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Eleanor Roosevelt"</code>
+    testString: assert(myRegex.test('Eleanor Roosevelt'), 'Your regex <code>myRegex</code> should return <code>true</code> for the string <code>"Eleanor Roosevelt"</code>');
+  - text: Your regex <code>myRegex</code> should return <code>false</code> for the string <code>"Franklin Rosevelt"</code>
+    testString: assert(!myRegex.test('Franklin Rosevelt'), 'Your regex <code>myRegex</code> should return <code>false</code> for the string <code>"Franklin Rosevelt"</code>');
```

Comment:
Hmm, would it be even simpler to grasp if we just did this: "match all variants of FDR's name": with middle name, without, with just "F."? I think it would. You then use the group with ? or * to also cover the no middle name case.

Reply:
Kinda, but to also drop Eleanor.

Reply:
So we have the check just for Franklin Roosevelt and other variations including his middle name?

Kind defeats the purpose, we wanted to have option grouping along with compulsory text. Like you had said: ` a(bb|cc)d` like that

Reply:
You just add Franklin Bozo Roosevelt to the negative tests and it's covered.

Reply:
Doing something, check it. Also, NO, I'm not writing Franklin Bozo Roosevelt 🤣

Reply:
Wait, doesn't the second test already kinda do that?

It allows for any length of a middle name

Reply:
Arbitrary ones, sure. I'll simplify it further: if you just ask to match F R and F D R, there won't be any need for an OR operator, as the regex would simply be F( D)? R. Tighter exercise.

Reply:
Looks fine now?

Reply:
I guess, but I'm also tired of repeating the same comments.